lib/rspec/core/configuration.rb

```diff
@@ -1421,6 +1421,22 @@ def warnings?
         $VERBOSE
       end
 
+      # Turns warnings into errors. This can be useful when
+      # you want RSpec to run in a 'strict' no warning situation.
+      #
+      # @example
+      #
+      #   RSpec.configure do |rspec|
+      #     rspec.raise_on_warning = true
+      #   end
+      def raise_on_warning=(value)
+        if value
+          RSpec::Support.warning_notifier = lambda { |message| raise message }
```

Comment:
I think this method would be better if this lambda was also extracted to a constant, that way we'd be at the same level of abstraction on the conditional. Thoughts?